### DISCUSSION TITLE
Is it desired that .setError() allows to set errors for fields that are not registered?

### DISCUSSION THREAD

**Discussion by mbrochh** [react-hook-form/react-hook-form#3537] [Q&A] [answered]
I lost lost three hours, trying to debug why my form refused to be submitted after it had received errors from the backend.

In the end, it turned out that my backend had reported an error for a field that is not actually part of my form, and now, even when I call `form.reset()` after my POST request is done, the form cannot be submitted again.

To reproduce this, just try this:

```
const formCtx = useForm()
useEffect(() => {
  formCtx.setError('bad_field', {message: 'uh oh'})
}, [])

const onSubmit = data => {
  console.log(data)
}

return (
<form onSubmit={formCtx.handleSubmit(onSubmit)}>
  <button type="submit">Submit</button>
</form>
)
```

You can press the button, but it won't go into the `onSubmit()` function.

**Comment by bluebill1049** [comment, marked as answer]
https://react-hook-form.com/api/#setError

<img width="947" alt="Screen Shot 2020-11-27 at 8 46 31 pm" src="https://user-images.githubusercontent.com/10513364/100435151-a3e6de00-30f1-11eb-8bc3-4f3aefc543d7.png">

`server` error has no association with the form, should be cleared with `clearErrors`.

**Reply by bluebill1049** [reply]
The reason for such design is so that each validation errors can/should have its own associated fields, while a custom error from the server should be reset by the (user/dev)'s action.

**Reply by mbrochh** [reply]
Great, thank you very much!

**Reply by bluebill1049** [reply]
Thanks for understanding, and sorry about your hours wasted. ❤️

**Reply by aryaniyaps** [reply]
@mbrochh if you've solved the problem,
may I please know where you are clearing your errors?

can't seem to figure that out.
thanks!

#4913
